### ISSUE TITLE
[ML] Deleting a forecast does not always work

### ISSUE THREAD

**Issue by sophiec20** [elastic/elasticsearch#74989]
**Elasticsearch version** (`bin/elasticsearch --version`):
Found in 7.14.0 BC1
Also reported in 7.10.1

**Description of the problem including expected versus actual behavior**:
Deleting forecasts does not always work.
This was initially raised when deleting `_all` forecasts where the API would respond as though it worked, but only some of the forecasts were deleted.

**Steps to reproduce**:

Test env: multi-node cluster with the gallery dataset using two multi-metric jobs, with model_plot enabled. 
count partition status 1h
sum(bytes) partition action 1h

1. Create a job with multiple forecasts as elastic superuser
2. Check stats to show count of forecasts `GET _ml/anomaly_detectors/my_job/_stats`
3. In Dev Tools, run `DELETE _ml/anomaly_detectors/my_job/_forecast/_all` - this returns `acknowledged: true`
4. Check stats to show count of forecasts `GET _ml/anomaly_detectors/my_job/_stats` - this shows some forecasts are still remaining.

(note - used a multi-node cluster, enabled model_plot and used larger dataset to make the steps more liable to timing issues. I did not check if this can be replicated with simpler configurations.) 

**Provide logs (if relevant)**:

Error logs do not indicate any problems. The following was logged even though only 3 of 4 forecasts where deleted.
```
[2021-07-06T14:20:06,155][INFO ][o.e.x.m.a.TransportDeleteForecastAction] [node2] Deleted forecast(s) [[GzIwfHoBCuniLktSdnsf, FzIvfHoBCuniLktS9nsd, EzIvfHoBCuniLktSqXtG, DzIvfHoBCuniLktSh3uS]] from job [my_job]
```

**Workarounds**

It was still possible to delete individual forecasts, although one zombie forecast did remain so this may not be a perfect workaround.

```
# To find forecast_id's
GET _sql?format=txt
{
  "query": """
  SELECT job_id, forecast_id, count(*) 
  FROM ".ml-anomalies-*"
  WHERE result_type = 'model_forecast_request_stats'
  OR result_type = 'model_forecast'
  GROUP BY forecast_id, job_id
  """
}

# To delete individual forecast:
DELETE _ml/anomaly_detectors/my_job/_forecast/8zIifHoBCuniLktS63rA
```

**Comment by przemekwitek** [comment]
FTR, I was able to reproduce the issue both manually and with integration test.

**Comment by przemekwitek** [comment]
Ok, I've found the issue:
In `TransportDeleteForecastAction.doExecute` method, we craft the `SearchRequest` in order to find the forecasts to be deleted. Unfortunately, we don't specify `size`, so the default of `10` is used.
Consequently, if there are more than 10 forecasts, not all of them will be deleted (only 10 will).

**Comment by przemekwitek** [comment]
> Ok, I've found the issue

That's one issue that only manifests itself when there are more than 10 forecasts.
In case described above (https://github.com/elastic/elasticsearch/issues/74989#issue-937997269) there were only 4 forecasts though. So there must be another issue.

**Comment by przemekwitek** [comment]
Ok, I was able to reproduce this issue with just 4 forecasts, just as Sophie did.
This time, it was the `max_docs` parameter of `DeleteByQueryRequest` being too small. In our code, we set `max_docs` explicitly to `10000` but when there are just a few forecasts with many partition field values, it is easy to have more than `10000` forecast documents in the results index.
Then DBQ does not delete all of them.
I think we should not set `max_docs` on DBQ as the default behavior of `max_docs` unset is "delete all" which is exactly what we need.
